config/v1/types_infrastructure.go

```diff
@@ -467,12 +467,29 @@ type BareMetalPlatformStatus struct {
 	// by components inside the cluster, like kubelets using the infrastructure rather
 	// than Kubernetes networking. It is the IP that the Infrastructure.status.apiServerInternalURI
 	// points to. It is the IP for a self-hosted load balancer in front of the API servers.
+	//
+	// Deprecated: Use APIServerInternalIPs instead
 	APIServerInternalIP string `json:"apiServerInternalIP,omitempty"`
 
+	// apiServerInternalIPs are the IP addresses to contact the Kubernetes API
+	// server that can be used by components inside the cluster, like kubelets
+	// using the infrastructure rather than Kubernetes networking. These are the
+	// IPs for a self-hosted load balancer in front of the API servers. In dual
+	// stack clusters this list contains two IPs otherwise only one.
+	APIServerInternalIPs []string `json:"apiServerInternalIPs,omitempty"`
```

Comment:
if both this and `APIServerInternalIP` are specified, which takes precedence?

also I think @deads2k has previously said we prefer to avoid omitempty so the api is more discoverable.

Reply:
also can we add some validation while we're introducing this new field?

Reply:
Thanks for your review @bparees. I added some validation to the new fields and removed the `omitempty` (wasn't aware that you prefer to avoid it...).
If both (`APIServerInternalIPs` and `APIServerInternalIP`) are set, then I think it depends on the user at last instance which to use. I added a deprecation notice to the old fields. Or what is the usual / better way to prevent from such behavior?

Reply:
> I think it depends on the user at last instance which to use. I added a deprecation notice to the old fields. Or what is the usual / better way to prevent from such behavior?

the most common pattern i've seen is that the older field takes precedence when set.  either by having consumers prefer the old field if it contains a value(until it goes away), or by having a controller that migrates the value from the old field into the new field.

Reply:
Ok, sorry I wasn't clear. With "user" I meant here the developer (us). As we use this struct mostly internally in on-prem context, we decide which takes precedence. In the installer, we followed the way to migrate the old field to the new field if it was set. Thanks for this hint!

Reply:
i still think the api doc needs to explain how the two fields interact

Reply:
Ok. I updated it and explained in the docs, which takes precedence if both are set.

Reply:
> In the installer, we followed the way to migrate the old field to the new field if it was set. 

so if the old field is set, you have logic that copies the value into the new field?  Does it stomp the existing value in the new field, if there is one?  Does that logic erase the value from the old field as part of the migration?

> Ok. I updated it and explained in the docs, which takes precedence if both are set.

thanks

Reply:
> so if the old field is set, you have logic that copies the value into the new field? Does it stomp the existing value in the new field, if there is one? Does that logic erase the value from the old field as part of the migration?

In the installer we only set this field. But for the fields in the install-config (not the ones declared in this PR, but similar (added new field with a slice for the VIPs)), we follow the described approach:
1. If the old field is set, but the new field is empty, we set the new field values based on the old values (-> migrate values from old field to new field)
2. If the old field is empty, we only take care on the new field
3. If both fields are set (new & old), we return an error (but this can be changed, e.g. that the installer tries to merge both fields (installer PR is WIP))

I hope this makes it a bit clearer

Reply:
that helps, thanks